op-node/metrics/metrics.go

```diff
@@ -289,6 +293,14 @@ func NewMetrics(procName string) *Metrics {
 			Name:      "stream_count",
 			Help:      "Count of currently connected p2p streams",
 		}),
+		PeerScores: factory.NewGaugeVec(prometheus.GaugeOpts{
+			Namespace: ns,
+			Subsystem: "p2p",
+			Name:      "peer_scores",
+			Help:      "Peer scoring",
+		}, []string{
+			"peerid",
+		}),
```

Comment:
I think this potentially opens up a DoS vector from my memory since a malicious actor could connect with thousands of different PeerIDs quite quickly, resulting in each one getting it's own metric (at least for prometheus each label value is effectively a new metric).

https://prometheus.io/docs/practices/naming/#labels has a warning about this.